app/src/main/java/bg/zahov/app/data/repository/SettingsRepositoryImpl.kt

```diff
@@ -6,25 +6,54 @@ import bg.zahov.app.data.interfaces.SettingsRepository
 import io.realm.kotlin.notifications.ObjectChange
 import kotlinx.coroutines.flow.Flow
 
+/**
+ * Repository implementation for managing user settings data.
+ *
+ * This singleton class interacts with the database layer via [RealmManager]
+ * to perform actions such as retrieving, updating, and resetting settings.
+ *
+ * @property realm Handles database operations related to settings through [RealmManager].
+ */
 class SettingsRepositoryImpl : SettingsRepository {
+
     companion object {
         @Volatile
         private var instance: SettingsRepositoryImpl? = null
+
+        /**
+         * Provides a thread-safe singleton instance of [SettingsRepositoryImpl].
+         *
+         * @return The singleton instance of [SettingsRepositoryImpl].
+         */
         fun getInstance() =
             instance ?: synchronized(this) {
-                instance ?: SettingsRepositoryImpl().also { instance = it } }
+                instance ?: SettingsRepositoryImpl().also { instance = it }
```

Comment:
Usually you don't need to do this. If you're doing something with potential race conditions perhaps the architecture needs updating. Not sure what the use case is.

Reply:
leaving it as it is